torch/csrc/jit/codegen/cuda/contiguity.cpp

```diff
@@ -512,17 +520,27 @@ void ContigIDs::handle(Merge* merge) {
 
   VectorOfUniqueEntries<IterDomain*> root_ids = root_ids_it->second;
 
+  bool is_indexing_pass = !ignore_consistent_ordering_;
+
   IterDomain* last_root = nullptr;
   for (auto root_id_i : c10::irange(root_domain_.size())) {
     auto root_id = root_domain_[root_id_i];
     if (root_ids.has(root_id)) {
       // ID found, remove it
       root_ids.erase(root_id);
-      // If the last id isn't contiguous that's fine, we can use the stride of
-      // the last iter domain to multiply the contig index.
-      if (!root_contiguity_[root_id_i] && !root_ids.empty()) {
-        // Otherwise this merge->out() isn't a contiguously indexable ID
-        return;
+      // If we're indexing:
+      // we could still potentially consider this ID linearly indexable, as we
+      // could multiple the index by the last root's stride.
+      //
+      // If we're computing predicates (ignore_consistent_ordering_==true),
+      // then we don't have this same constraint, we can just ignore
+      // contiguity of the roots all together.
+      if (!root_contiguity_[root_id_i] && is_indexing_pass) {
+        // TODO: This didn't error when I removed "!ignore_consistent_ordering_"
+        // is it really needed?
```

Comment:
Do you still want to keep this comment? It should actually not be necessary. If this is used for indexing, `is_indexing_pass` is true, so nothing should change. If used for predicates, `!root_contiguity_` should be always true, so `is_indexing_pass` doesn't matter.

I actually like the current code as it's easier to understand the underlying logic.